### ISSUE TITLE
Unable to use messagesPerField existsError and get

### ISSUE THREAD

**Issue by Romcol** [InseeFrLab/keycloakify#40]
Hello,

I'm glad that you introduced `messagesPerField.exists` and `messagesPerField.get` in the new keycloakify v2.4.0 !
Though, I'm unable to use it. 
Looking at the code, it seems that if I have no `profile.attributes` in my configuration. I don't know if that's necessary as I would like to use it on classic fields such as `userLabel`, `username`, `email`, `firstName`, `lastName`, `password` and `password-confirm`.

I'm not sure if this is a bug or whether it is intended.
I would be happy if you could update the lib for it to iterate on those values at least.

Cheers!

**Comment by Romcol** [comment]
Cheers @garronej ! It seems to work now :) I'll let you know if I have more problems regarding this

**Comment by blzzz** [comment]
Hey @Romcol ,
How exactly did you manage to add custom fields via `profile.attributes`? Using  `messagesPerField.existsError("user.attributes.dateOfBirth")` for example throws an error as the field is not defined by default. I cannot find out how to extend the list of rendered fields – but it seems somehow doable via `profile.attributes`, right?

**Comment by Romcol** [comment]
Hi @blzzz 

You need to use the name directly as such `messagesPerField.existsError("dateOfBirth")`. You can check the source code of the page for better understanding :)

Good luck

**Comment by garronej** [comment]
Hi @loki344,  
I understand your problem... 
If you want I can add `"user.attributes.country"` to [the list](https://github.com/InseeFrLab/keycloakify/blob/020823e933be609a98766ded8f70785612515d7d/src/bin/build-keycloak-theme/generateFtl/ftl_object_to_js_code_declaring_an_object.ftl#L13-L17) if it can help you out ?

**Comment by loki344** [comment]
Hey @garronej 
Yes this would be perfect! We added "user.attributes.dateOfBirth", "user.attributes.country" and "user.attributes.acceptedTermsAndConditions".

Thank you for your help

**Comment by garronej** [comment]
Hi @loki344,  
I added those fields and released a new version (5.9.0) to enable you do move forward with this but note that in the upcoming Keycloakify v6 I will remove them and you would have instead to declare them in the `package.json` like:  
```json
    "keycloakify": {
        "extraPages": [
            "my-extra-page-1.ftl",
            "my-extra-page-2.ftl"
        ],
        "extraFields": [ 
               "user.attributes.dateOfBirth", 
               "user.attributes.country", 
               "user.attributes.acceptedTermsAndConditions" 
         ]
    },
```  

Maybe it's best if you add them right away since it wont be mentioned in the v5 to v6 migration guide.

**Comment by loki344** [comment]
Wow thank you very much for your support!!

**Comment by loki344** [comment]
> Hi @loki344, I added those fields and released a new version (5.9.0) to enable you do move forward with this but note that in the upcoming Keycloakify v6 I will remove them and you would have instead to declare them in the `package.json` like:
> 
> ```json
>     "keycloakify": {
>         "extraPages": [
>             "my-extra-page-1.ftl",
>             "my-extra-page-2.ftl"
>         ],
>         "extraFields": [ 
>                "user.attributes.dateOfBirth", 
>                "user.attributes.country", 
>                "user.attributes.acceptedTermsAndConditions" 
>          ]
>     },
> ```
> 
> Maybe it's best if you add them right away since it wont be mentioned in the v5 to v6 migration guide.

Will this version be available via https://www.npmjs.com/package/keycloakify ?

**Comment by garronej** [comment]
Yes sorry the CI workflow that automatically release faild for some reason... I'm looking into it.

**Comment by garronej** [comment]
v5.9.0 is now actually released on NPM.

**Comment by loki344** [comment]
You're the man! Greetings from Switzerland!

**Comment by garronej** [comment]
You are wellcome!  
The patch literally landed lol 
<img width="892" alt="image" src="https://user-images.githubusercontent.com/6702424/182505764-0eed11bb-e95d-4338-a98e-3de36f3ead03.png">

**Comment by garronej** [comment]
> > Hi @loki344, I added those fields and released a new version (5.9.0) to enable you do move forward with this but note that in the upcoming Keycloakify v6 I will remove them and you would have instead to declare them in the `package.json` like:
> > ```json
> >     "keycloakify": {
> >         "extraPages": [
> >             "my-extra-page-1.ftl",
> >             "my-extra-page-2.ftl"
> >         ],
> >         "extraFields": [ 
> >                "user.attributes.dateOfBirth", 
> >                "user.attributes.country", 
> >                "user.attributes.acceptedTermsAndConditions" 
> >          ]
> >     },
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Maybe it's best if you add them right away since it wont be mentioned in the v5 to v6 migration guide.
> 
> Will this version be available via https://www.npmjs.com/package/keycloakify ?

After months, this build option has finally been released.

**Comment by garronej** [comment]
@Romcol  @dragonworr Since Keycloakify 7.13, it's no longer needed to provide `extaFields`.  
Keycloakify predicts the fieldNames from usage.  https://docs.keycloakify.dev/build-options#customuserattributes
